internal/controllers/synthesis/lifecycle.go

```diff
@@ -131,17 +126,21 @@ func (c *podLifecycleController) Reconcile(ctx context.Context, req ctrl.Request
 		}
 
 		// Remove the finalizer when all pods and slices have been deleted
-		if len(sliceList.Items) > 0 || len(pods.Items) > 0 {
-			logger.V(1).Info(fmt.Sprintf("refusing to remove composition finalizer because %d associated resource slices and %d pods still exist", len(sliceList.Items), len(pods.Items)))
-			return ctrl.Result{}, nil // some resources still exist
+		if comp.Status.CurrentState != nil && (!comp.Status.CurrentState.Reconciled) || comp.Status.CurrentState.ObservedCompositionGeneration != comp.Generation {
+			logger.V(1).Info("refusing to remove composition finalizer because some resources have not yet been cleaned up")
```

Comment:
We can use the aggregated status signal when cleaning up instead of the old resource slice index.